### PR TITLE
Add activemq config for fcrepo

### PR DESCRIPTION
**Pull request by jonathangreen**
## Ticket
https://github.com/Islandora-CLAW/CLAW/issues/711

## Issue
Fcrepo wasn't coming up because of two activemq, embedded and external. Same fix as for the vagrant box we need to configure fedora to not start activemq broker.

## Testcase
Run playbook and make sure fcrepo comes up.

### CODE REVIEW

**Comment by jonathangreen** [comment]
@whikloj @dannylamb @ajs6f 

ran into this while testing the crayfish pull. if anyone has time to review and merge would be most appreciated.

File: roles/internal/fcrepo/defaults/main.yml [hunk 1]
```diff
@@ -3,3 +3,5 @@ fcrepo_user: tomcat8
 fcrepo_data_dir: /var/lib/tomcat8/fcrepo4-data
 fcrepo_war_path: "{{ tomcat8_home }}/webapps/fcrepo.war"
 fcrepo_home_dir: /opt/fcrepo
+fcrepo_activemq_template: activemq.xml.j2
+fcrepo_tomcat_config_dir: "{{ tomcat8_home }}/webapps/fcrepo/WEB-INF/classes/config/"
```
Comment (ajs6f) [inline roles/internal/fcrepo/defaults/main.yml]: It's generally not best practice to put config like this inside the filesystem space of the webapp itself... maybe someplace inside `/usr/local` or `/etc` or the like?

File: roles/internal/fcrepo/templates/activemq.xml.j2 [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    http://activemq.apache.org/schema/core http://activemq.apache.org/schema/core/activemq-core.xsd">
+
+    <context:property-placeholder/>
+
+    <!--
+        The <broker> element is used to configure the ActiveMQ broker.
+    -->
+    <broker xmlns="http://activemq.apache.org/schema/core" brokerName="localhost">
+      <persistenceAdapter>
+        <kahaDB directory="${fcrepo.activemq.directory:ActiveMQ/kahadb}"/>
```
Comment (ajs6f) [inline roles/internal/fcrepo/templates/activemq.xml.j2]: Are we setting `fcrepo.activemq.directory` somewhere?

File: roles/internal/fcrepo/defaults/main.yml (hunk 1 above)
Comment (jonathangreen) [reply roles/internal/fcrepo/defaults/main.yml]: I agree with you 100%. Fedora itself puts a `activemq.xml` file at this path, that I am trying to override here. Is there some way to pass a path for this config to fedora so it reads the alternative configuration? I'm not sure how I would do that.

File: roles/internal/fcrepo/templates/activemq.xml.j2 (hunk 2 above)
Comment (jonathangreen) [reply roles/internal/fcrepo/templates/activemq.xml.j2]: I assume its being set in the fedora configs somewhere. I stole this line directly from the activemq config that ships with fcrepo. 


https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-configs/src/main/resources/config/activemq.xml#L78-L80

Comment (ajs6f) [reply roles/internal/fcrepo/templates/activemq.xml.j2]: Okay, cool, then it's not CLAW's problem.

File: roles/internal/fcrepo/defaults/main.yml (hunk 1 above)
Comment (ajs6f) [reply roles/internal/fcrepo/defaults/main.yml]: Jeez, I can't believe Fedora does that. No, I can totally believe that. If that's the way that Fedora does it, I guess we're probably better off leaving it alone.

**Comment by ajs6f** [comment]
I'll wait a bit through today in case anyone else has comments, then I'll merge this. It's pretty much a no-brainer.

File: roles/internal/fcrepo/defaults/main.yml (hunk 1 above)
Comment (whikloj) [reply roles/internal/fcrepo/defaults/main.yml]: I think we reverted my work on Fedora to make a single configuration file that you could override with a JAVA_OPT. I'll look back and see.

Comment (whikloj) [reply roles/internal/fcrepo/defaults/main.yml]: Ok, I think we can put the `activemq.xml` anywhere and set use this https://github.com/fcrepo4/fcrepo4/blob/fcrepo-4.7.4/fcrepo-webapp/src/main/resources/spring/jms.xml#L18-L19

Comment (whikloj) [reply roles/internal/fcrepo/defaults/main.yml]: So I'm not sure how we do this in Ansible, but we need to add
`-Dfcrepo.activemq.configuration=file:/blah/blah/activemq.xml`
to
https://github.com/Islandora-Devops/claw-playbook/blob/master/inventory/vagrant/group_vars/tomcat.yml#L9-L22

Comment (jonathangreen) [reply roles/internal/fcrepo/defaults/main.yml]: Cool. Thanks @whikloj I'll update the playbook.

**Comment by jonathangreen** [comment]
@whikloj @ajs6f updated the path for the config file.

**Comment by ajs6f** [comment]
I know @dannylamb is out this week, so can I assume that everyone who wants to has chimed in on this PR?

**Pull request merged**